FIX reshape the predicitons array shape in IterativeImputer (#19409)
scikit-learn/scikit-learn: doc/whats_new/v1.0.rst

  :class:`~sklearn.semi_supervised.LabelPropagation`.
  :pr:`19271` by :user:`Zhaowei Wang <ThuWangzw>`.

:mod:`sklearn._iterative.py`
.................................

> - |Fix| Fix ValueError triggered when setting the estimator as PLSRegression() in 
>   :class: `sklearn._iterative.py: func: _impute_one_feature`

Reviewer: Maybe a higher-level entry might be clearer to users?
```suggestion
- |Fix| :class:`cross_decomposition.PLSRegression` can now be used as an estimator for 
  :class:`impute.IterativeImputer`.
```

Reviewer 2: And we will need a non-regression test that should ensure that it is actually working in this PR while failing in the branch `main`

Reviewer: I got back to the original issue, and it looks like some issues (like [`SkuaD01/scikit-learn#1`](https://github.com/SkuaD01/scikit-learn/pull/1/files)) implemented a test for this.

@isaack-mungui and @SkuaD01: you probably could work together to have your changes integrated in `mean`. 🙂

Author: @jjerphan Just to gain clarity, should I integrate changes in `mean` or follow through with @glemaitre's requested changes first?

Reviewer: You can proceed with @glemaitre's request (I misspelled `main` and wrote `mean`).

Author: Got it. Thanks!
